src/main/java/tigase/xmpp/impl/SaslAuth.java

```diff
@@ -211,7 +216,7 @@ public void process(final Packet packet, final XMPPResourceConnection session, f
 														  session.getDomain().getVhost().getDomain());
 						}
 
-						if (bruteForceLocker.isEnabled(session) &&
+						if (bruteForceLocker != null && bruteForceLocker.isEnabled(session) &&
```

Comment:
The same as in comment for `SaslAuth2` about new method. Maybe it would be worth to add this new method to the new class `AbstractAuthWithBruteForceLockPreprocessor` and rebase those classes to use this new class?

Reply:
Or maybe add BruteForceLocker to `AbstractAuthPreprocesor` and make that class a "wrapper" for using BruteForceLocker?

Reply:
I moved `BruteForceLocker` checking to `AbstractAuthPreprocesor` as it made the most sense (it should be used by all auth plugins). While working on it, I also refactored most of the common parts of `SaslAuth` and `SaslAuth2` to `SaslAuthAbstract`.

Reply:
OK